File: 02-programacion-basica/04-condicionales/scala/soluciones/RetoEvaluarCurso.scala
import scala.io.StdIn.readLine
import scala.util.Try

@main def retoEvaluarCurso(): Unit =
  val nota = Try(readLine("Nota (0-20): ").trim.toInt).toOption
  val asistencia = Try(readLine("Asistencia (0-100): ").trim.toInt).toOption

  // El orden de las condiciones evita clasificar datos inválidos.
  val resultado = (nota, asistencia) match
    case (Some(n), Some(a)) if n < 0 || n > 20 || a < 0 || a > 100 => "Datos fuera de rango"
    case (Some(_), Some(a)) if a < 70 => "Desaprobado por asistencia"
    case (Some(n), Some(_)) if n < 11 => "Desaprobado por nota"
    case (Some(_), Some(_)) => "Aprobado"
    case _ => "Debe ingresar números enteros"

  println(resultado)

File: 02-programacion-basica/04-condicionales/java/soluciones/RetoEvaluarCurso.java
import java.util.Scanner;

public class RetoEvaluarCurso {
    public static void main(String[] args) {
        int nota;
        int asistencia;

        try (Scanner scanner = new Scanner(System.in)) {
            System.out.print("Nota (0-20): ");
            nota = Integer.parseInt(scanner.nextLine().trim());
            System.out.print("Asistencia (0-100): ");
            asistencia = Integer.parseInt(scanner.nextLine().trim());
        } catch (NumberFormatException e) {
            System.out.println("Debe ingresar números enteros");
            return;
        }

        // Primero se valida el dominio para no clasificar entradas inválidas.
        if (nota < 0 || nota > 20 || asistencia < 0 || asistencia > 100) {
            System.out.println("Datos fuera de rango");
        } else if (asistencia < 70) {
            System.out.println("Desaprobado por asistencia");
        } else if (nota < 11) {
            System.out.println("Desaprobado por nota");
        } else {
            System.out.println("Aprobado");
        }
    }
}
